labs/query-enriched-event-data-with-spark/README.md

```diff
@@ -7,59 +7,87 @@ Spark is a powerful, widely-adopted engine for data processing. It's easy to run
 In this Lab, we'll assume you're working with a standalone Spark cluster running on your computer.  However, this and other Lab notebooks can be modified to work with remote Spark clusters as well.
 
 
-## Running PySpark locally with Docker
+## Running this notebook
 
-The simplest way to get started with PySpark is to run it in a [Docker](https://www.docker.com/) container.  You use Docker to run the Lab notebook with a single command:
+There are several ways to run this notebook locally:
+- Using the `run.sh` script
+- Using [Docker](https://www.docker.com/) with the `run-docker.sh` script
+- Manually, using the `conda` CLI
 
-```sh
-docker run -it --rm \
-    -p 8888:8888 \
-    -v $(pwd):/home/jovyan \
-    jupyter/pyspark-notebook \
-    jupyter lab query_enriched_event_data_with_spark.ipynb
-```
+### Running the notebook with `run.sh`
+
+You can use the `run.sh` script to build your environment and run this notebook with a single command.
+
+#### Prerequisite: conda (version 4.4+)
+
+[Anaconda]: https://www.anaconda.com/distribution/
+[Miniconda]: https://docs.conda.io/en/latest/miniconda.html
+
+You can install the `conda` CLI by installing [Anaconda] or [Miniconda].
 
-Docker makes it easy to get started with PySpark, but it adds overhead and may require [additional configuration](https://docs.docker.com/config/containers/resource_constraints/) to handle large workloads.  
+#### Running Jupyter Lab
 
-## Running Spark locally on the JVM
+This lab directory contains a handy script for building your conda environment and running Jupyter Lab.  To run it, simply use
+
+```sh
+bash bin/run.sh
+```
 
-Running Spark "natively" on the [Java Virtual Machine](https://en.wikipedia.org/wiki/Java_virtual_machine) takes a bit more work, but is generally preferred.
+That's it, you're done!
```

Comment:
It took me like a second to realize that once Jupyter was running, I should focus on the query_enriched_event_data_with_spark `ipynb` file. I'd be curious if this should be called out specifically or if anyone who knows what Jupyter is will know where to go